File: compat/src/main/java/android/content/pm/ArchiveInfo.java
package android.content.pm;

import android.os.Parcel;
import android.os.Parcelable;

import androidx.annotation.NonNull;

import java.io.File;

public class ArchiveInfo implements Parcelable {
    String path;
    String packageName;
    String originating;

    public File getPackageFile() {
        return new File(path);
    }

    public String getPackageName() {
        return packageName;
    }

    public String getOriginating() {
        return originating;
    }

    public ArchiveInfo(@NonNull String path, @NonNull String packageName, String originating) {
        this.path = path;
        this.packageName = packageName;
        this.originating = originating;
    }

    public ArchiveInfo(@NonNull File path, @NonNull String packageName, String originating) {
        this.path = path.getPath();
        this.packageName = packageName;
        this.originating = originating;
    }

    protected ArchiveInfo(Parcel in) {
        path = in.readString();
        packageName = in.readString();
        originating = in.readString();
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(path);
        dest.writeString(packageName);
        dest.writeString(originating);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public static final Creator<ArchiveInfo> CREATOR = new Creator<>() {
        @Override
        public ArchiveInfo createFromParcel(Parcel in) {
            return new ArchiveInfo(in);
        }

        @Override
        public ArchiveInfo[] newArray(int size) {
            return new ArchiveInfo[size];
        }
    };
}

File: compat/src/main/kotlin/dev/sanmer/hidden/compat/content/ArchiveInfo.kt
package dev.sanmer.hidden.compat.content

import android.content.pm.PackageInfo
import android.os.Parcel
import android.os.Parcelable
import dev.sanmer.hidden.compat.utils.readParcelable
import java.io.File

data class ArchiveInfo(
    private val archiveFilePath: String,
    val originatingPackageName: String,
    val archivePackageInfo: PackageInfo,
) : Parcelable {
    val archiveFile: File get() = File(archiveFilePath)

    constructor(parcel: Parcel) : this(
        archiveFilePath = checkNotNull(parcel.readString()),
        originatingPackageName = checkNotNull(parcel.readString()),
        archivePackageInfo = checkNotNull(parcel.readParcelable(PackageInfo::class.java))
    )

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(archiveFilePath)
        parcel.writeString(originatingPackageName)
        parcel.writeParcelable(archivePackageInfo, flags)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<ArchiveInfo> {
        override fun createFromParcel(parcel: Parcel): ArchiveInfo {
            return ArchiveInfo(parcel)
        }

        override fun newArray(size: Int): Array<ArchiveInfo?> {
            return arrayOfNulls(size)
        }
    }
}